### PR TITLE
Fix/xss issues on data attributes

### LINKED ISSUE #26625: XSS possible in collapse data-parent attribute
XSS possible in collapse data-parent attribute
`data-parent=""`
Bootstrap 4.1.1 in combination with JQuery 3.3.1

### LINKED ISSUE #26628: XSS possible in data-container property of tooltip
XSS possible in tooltip data-container attribute
`data-container=""`
Bootstrap 4.1.1 in combination with JQuery 3.3.1

### LINKED ISSUE #27044: tooltip - XSS on data-viewport attribute
found in bootstrap 3.3.7

`<a href="#" data-toggle="tooltip" data-viewport="" title="Hooray!">Hover over me</a>`
https://jsbin.com/qipirurise/edit?html,output

Win 7 x64
Chrome 67.0.3396.99
Firefox 61.0.1 (64-Bit)

### LINKED ISSUE #27045: affix - XSS on target config
found in bootstrap 3.3.7

`$('.js-affixed-element-top').affix({
    offset: {
      top: $('.js-page-header').outerHeight(true) - 10
    , bottom: $('.js-footer').outerHeight(true) + 10
    },
    target: ''
  })`

https://jsbin.com/palokaxina/edit?html,output

Win 7 x64
Chrome 67.0.3396.99
Firefox 61.0.1 (64-Bit)

### PR DESCRIPTION
**Pull request by don-spyker**
fix of xss issues that have been fixed in v4.x already + fix of two additional xss issues

Fixes #26625 / CVE-2018-14040
Fixes #26628 / CVE-2018-14042
Fixes #27044 / CVE-2018-20676
Fixes #27045 / CVE-2018-20677

### CODE REVIEW

File: js/affix.js [hunk 1]
```diff
@@ -16,7 +16,9 @@
   var Affix = function (element, options) {
     this.options = $.extend({}, Affix.DEFAULTS, options)
 
-    this.$target = $(this.options.target)
+    var target = this.options.target === Affix.DEFAULTS.target ? $(this.options.target) : $(document).find(this.options.target)
```
Comment (Johann-S) [inline js/affix.js]: There is no need for your ternary here: `$(document).find(this.options.target)` is enough

**Review by Johann-S — CHANGES_REQUESTED** [review]
You need unit test for each plugins you changed

File: js/affix.js (hunk 1 above)
Comment (don-spyker) [reply js/affix.js]: I tried that first, but that broke the functionality as the default is "window" and $(document).find('window') didn't work while $(window) evaluates correctly.

Comment (Johann-S) [reply js/affix.js]: Ok I understand :+1:

**Pull request merged**